apm-agent-plugins/apm-process-plugin/src/test/java/co/elastic/apm/agent/process/ProcessHelperTest.java

```diff
@@ -168,19 +168,32 @@ void waitForWithTimeoutDoesNotEndProcessSpan() {
 
         helper.doEndProcess(process, true);
         assertThat(storageMap).isEmpty();
+
+        Span span = getFirstSpan();
+        assertThat(span.getOutcome()).isEqualTo(Outcome.SUCCESS);
     }
 
     @Test
     void destroyWithoutProcessTerminatedShouldEndSpan() {
         Process process = mock(Process.class);
         verifyNoMoreInteractions(process); // we should not even use any method of process
 
+        // we have to opt-in to allow unknown outcome
+        reporter.checkUnknownOutcome(false);
```

Comment:
@eyalkoren external process execution is one case where we might not always know the outcome if the process hasn't yet terminated. But I have to admit that it's quite likely to be a very narrow corner case.

Reply:
I think this is a fair exception to make - if we never report `unknown` to terminated processes and always report `unknown` to unterminated processes it sounds reasonable.